src/UnicycleGenerator.cpp

```diff
@@ -619,6 +661,17 @@ std::shared_ptr<FeetCubicSplineGenerator> UnicycleGenerator::addFeetCubicSplineG
 
 }
 
+std::shared_ptr<FeetMinimumJerkGenerator> UnicycleGenerator::addFeetMinimumJerkGenerator()
+{
+    std::lock_guard<std::mutex> guard(m_pimpl->mutex);
+
+    if (m_pimpl->feetMinimumJerkGenerator == nullptr) {
+        m_pimpl->feetMinimumJerkGenerator.reset(new FeetMinimumJerkGenerator());
```

Comment:
Instead of using 
```cpp
m_pimpl->feetMinimumJerkGenerator.reset(new FeetMinimumJerkGenerator());
```
You may use 
```cpp
m_pimpl->feetMinimumJerkGenerator = std::make_shared<FeetMinimumJerkGenerator>();
```
[Here](https://stackoverflow.com/questions/20895648/difference-in-make-shared-and-normal-shared-ptr-in-c) and [here](https://stackoverflow.com/questions/36477022/boost-initializing-shared-pointer-reset-vs-make-shared/36477600) you can find some reasons why this is in general better. In simple terms it just a matter of _heap-allocation_. Using `std::make_shared` only one heap-allocation is performed while, on the other hand, `new FeetMinimumJerkGenerator()` invokes a heap-allocation for the managed data and the `.reset( )` method performs another one.

Reply:
Unfortunately it is not possible. The constructors are ``private`` and can be allocated only because ``UnicyleGenerator`` is a ``friend``. As a consequence ``std::make_shared`` fails is allocating those objects, while with ``new`` we circumvent this issue.